src/input.service.ts

```diff
@@ -44,7 +44,8 @@ export class InputService {
         let keyChar = String.fromCharCode(keyCode);
         let selectionStart = this.inputSelection.selectionStart;
         let selectionEnd = this.inputSelection.selectionEnd;
-        this.rawValue = this.rawValue.substring(0, selectionStart) + keyChar + this.rawValue.substring(selectionEnd, this.rawValue.length);
+       // this.rawValue = this.rawValue.substring(0, selectionStart) + keyChar + this.rawValue.substring(selectionEnd, this.rawValue.length);
+        this.rawValue =   this.rawValue+ keyChar;
```

Comment:
Here we may have a problem, the commented code works when the user selects a piece of the number and try to replace it.

Example: We have **9.999,99**, select with the cursor the last two **99**, and types **7**... The result would be **999,97**. But after the comment, the results become **9.999,97**.

I suggest you create a function with the commented code, and check if the **selectionStart** and **selectionEnd** are equal before ignoring completely the substring.

```suggestion
      this.rawValue = getCurrentRawValue();
      
      getRawValueBySelection(selectionStart, selectionEnd, keyChar) {
        // "this.rawValue + keyChar" should be returned only if both 
        // selectionStart and selectionEnd are equal 0
        // that means that the cursor is at the start of the number
        ...
      }
```

Reply:
Ok, i will look into it,